dsp/utils/settings.py

```diff
@@ -39,7 +39,10 @@ def __new__(cls):
 
     @property
     def config(self):
-        return self.stack_by_thread[threading.get_ident()][-1]
+        thread_id = threading.get_ident()
+        if thread_id not in self.stack_by_thread:
+            self.stack_by_thread[thread_id] = [self.main_stack[0].copy()]
```

Comment:
shouldn't this be main_stack[-1]?

Reply:
same below etc, not 100% sure but seems so

Reply:
ah yes if we want new threads to start with the latest configuration, self.main_stack[-1] would be the correct fix. however, if we wanted to maintain the original configuration for every thread, i defaulted it to the current setup but can change as per our use case.

Reply:
It should definitely start from the parent thread, not from scratch